### ISSUE TITLE
[Feature Request] react-native-google-cast integration

### ISSUE THREAD

**Issue by petrbela** [DoubleSymmetry/react-native-track-player#1421]
Hey guys!

I'm the maintainer of react-native-google-cast package. I know from comments that some of the users are already using track player and trying to add google cast support. I noticed track player has a wip package for basic google cast support but I was thinking whether it would make more sense to try to come up with some integration layer between the two, as I can imagine standalone support for the whole suite of features that Google Cast SDK provides is probably out of scope for you.

Do you have any thoughts on possible plugin/extension API to the track player to enable integrating with other libraries?

**Comment by DennisdeWitNL** [comment]
+1. Would love that. I am using your module seperately currently. Luckily, both modules can read variables. So I first init GoogleCast with the JSON info, and then do the same with RNTP. If it could get any easier, why wouldn't we? :)

**Comment by martinmidtsund** [comment]
Yes, this would be great. I have noticed an issue with RNTP and Google Cast where the RNTP application is put to sleep while casting because the app is no longer playing the sound, and thus are not able to send e.g. analytics while playing audio, not tested this myself but ref to issue is here: https://github.com/react-native-google-cast/react-native-google-cast/issues/270 

This could perhaps be solved with a custom receiver as suggested here: https://github.com/react-native-google-cast/react-native-google-cast/issues/269

Anyways, a closer integration would be awesome! 👏

**Comment by jspizziri** [comment]
@petrbela @DennisdeWitNL @martinmidtsund ,

Thanks for your feedback and the interest! I'm not super familiar with Google Cast in RNTP or in react-native-google-cast. However the idea sounds like a good one from my perspective.

@dcvz would be interested in your take on this.

**Comment by jspizziri** [comment]
Given there's no one to champion this at the moment, I'm going to add it to the feature backlog and close the issue. We can revisit it again in the future.

**Comment by DigenPrimeApps** [comment]
@jspizziri any solution on this feature?
